GSA_Adapter/CRUD/Read.cs

```diff
@@ -148,24 +128,22 @@ public List<IMaterialFragment> ReadMaterials(List<string> ids = null, bool inclu
             else
                 materials = matArr.Where(x => ids.Contains(x.Split(',')[1])).Select(x => Convert.FromGsaMaterial(x)).Where(x => x != null).ToList();
 
-            if (includeStandard)
-                materials.AddRange(GetStandardGsaMaterials());
+            return materials;
+        }
+
+        /***************************************************/
 
-            for (int i = 0; i < materials.Count(); i++)
+        public Dictionary<string, IMaterialFragment> ReadMaterialDictionary(List<string> ids = null, bool includeStandard = false)
+        {
+            Dictionary<string, IMaterialFragment> materials = GetCachedOrReadAsDictionary<string, IMaterialFragment>(ids);
+            if (includeStandard)
             {
-#if GSA_10_1
-                string key = matArr[i].Split(("_.").ToCharArray())[1] + ":" + materials[i].GSAId();
```

Comment:
@alelom Regarding your comment - It is moving the line here to the MaterialConvert method.
It does slightly change the GSAId for materials being read in GSA 10, but I think it most likely is for the better.

Before this is merged though, I will test through that this does not have any strange side effects for thigns like Updating materials.